src/security/builtin_plugins/authentication/src/authentication.c

```diff
@@ -216,21 +220,19 @@ typedef struct {
 } validity_cb_info;
 
 
+static int compare_relation (const void *va, const void *vb);
+
+const ddsrt_avl_treedef_t relations_treedef =
+  DDSRT_AVL_TREEDEF_INITIALIZER (offsetof (struct IdentityRelation, avlnode), offsetof (struct IdentityRelation, remoteIdentity), compare_relation, 0);
+
 static bool
 security_object_valid(
     SecurityObject *obj,
     SecurityObjectKind_t kind)
 {
     if (!obj) return false;
     if (obj->kind != kind) return false;
-    if (kind == SECURITY_OBJECT_KIND_IDENTITY_RELATION) {
-        IdentityRelation *relation = (IdentityRelation *)obj;
-        if (!relation->localIdentity || !relation->remoteIdentity || (ddsrt_address)obj->handle != (ddsrt_address)relation->localIdentity) {
-            return false;
-        }
-    } else if ((ddsrt_address)obj->handle != (ddsrt_address)obj) {
-        return false;
-    }
+    if ((ddsrt_address)obj->handle != (ddsrt_address)obj) return false;
```

Comment:
My apologies for overlooking the ``ddsrt_address`` in some previous review, but there really is no need for it to exist. This should just use ``uintptr_t``.